tests/storage/hotplug.go

```diff
@@ -1019,10 +1026,91 @@ var _ = SIGDescribe("Hotplug", func() {
 				addVolumeFunc(vmi.Name, vmi.Namespace, volumeName, dv.Name, v1.DiskBusSCSI, false, "")
 				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
-				VerifyVolumeAndDiskVMIAdded(virtClient, vmi, volumeName)
+				verifyVolumeAndDiskVMIAdded(virtClient, vmi, volumeName)
 				verifyVolumeStatus(vmi, v1.VolumeReady, "", volumeName)
 			})
 		})
+
+		Context("disk mutating sidecar", func() {
+			const (
+				hookSidecarImage     = "example-disk-mutation-hook-sidecar"
+				sidecarContainerName = "hook-sidecar-0"
+				newDiskImgName       = "kubevirt-disk.img"
+			)
+
+			var (
+				vm *v1.VirtualMachine
+				dv *cdiv1.DataVolume
+				sc string
+			)
+
+			renderSidecar := func() map[string]string {
+				return map[string]string{
+					"hooks.kubevirt.io/hookSidecars":         fmt.Sprintf(`[{"image": "%s/%s:%s", "imagePullPolicy": "IfNotPresent"}]`, flags.KubeVirtUtilityRepoPrefix, hookSidecarImage, flags.KubeVirtUtilityVersionTag),
+					"diskimage.vm.kubevirt.io/bootImageName": newDiskImgName,
+				}
+			}
+
+			BeforeEach(func() {
+				if !libstorage.HasCDI() {
+					Skip("Skip tests when CDI is not present")
+				}
+				var exists bool
+				sc, exists = libstorage.GetRWXBlockStorageClass()
+				if !exists {
```

Comment:
So only in one of the 3 entries of this table are you using block, and none of them actually require RWX, would it make sense to pass the function that looks up a particular storage class into the table, and then call that first thing to find the storage class and use it in the test.

Something like this:
```go
DescribeTable("should be able to add and remove volumes", func(addVolumeFunc addVolumeFunction, removeVolumeFunc removeVolumeFunction, storageClassFunc func()(string, error), volumeMode corev1.PersistentVolumeMode, vmiOnly bool) {
  sc, exists := storageClassFunc()
  if !exists {
    Skip("Skip test when appropriate storage class is not available")
  }
  ...
}
```
And in the entry pass in `libstorage.GetRWOFileSystemStorageClass` and `libstorage.GetRWOBlockStorageClass` for the various permutations.

Reply:
My bad, I used the `should add/remove volume` test as reference and saw that we use `GetRWXBlockStorageClass` there. I assumed that was necessary. I can change that to use specific classes like you suggested.

Reply:
We probably need to revisit those tests to make sure we are not using stuff we don't need. We should try and use the minimum needed for the test, so it can run against as many storage providers as possible.